File: bubble/bubble.c
#include <time.h>
#include <stdlib.h>
#include <stdio.h>

#include <FPT.h>
int main(){
                  clock_t begin,end;

	int length = 10000;
	double arr[length];
	int i, j;

	srand(time(NULL));
                  int w;
      for(w=0;w<100;w++){
            begin = clock();
	for(i=0;i<length;i++){
		arr[i] = rand();
	}

	
      int swapped = 1;

      j = 0;

      int tmp;

      while (swapped) {

            swapped = 0;

            j++;

            for (i = 0; i < length - j; i++) {

                  if (arr[i] > arr[i + 1]) {

                        tmp = arr[i];

                        arr[i] = arr[i + 1];

                        arr[i + 1] = tmp;

                        swapped = 1;

                  }

            }

      }
                  end = clock();

      double dur = (double)(end-begin)/CLOCKS_PER_SEC;
      printf("%lf\n", dur);
}

}

File: bubble/bubble.rb

n = 10000

for k in 0..99
      start = Time.now
      swapped = true
      nums = Array.new(n){rand{0..1000000}}
      j=0
      while swapped
            swapped = false
            j = j + 1
            for i in 0..(n-j-1)
                  if nums[i] > nums[i+1]
                        temp = nums[i]
                        nums[i] = nums[i+1]
                        nums[i+1] = temp
                        swapped = true
                  end
            end
      end
      fin = Time.now
      puts fin - start
end